### PR TITLE
handle non-existing roots

### PR DESCRIPTION
**Pull request by TonioF**
Avoids errors when a file system data store is created with a root that does not yet exist. The root will be created the first time that data is written to the store, it will be ignored beforehand.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## Changes in 0.9.2
```
Comment (AliceBalfanz) [inline CHANGES.md:1]:
```suggestion
## Changes in 0.9.2 (in development)
```

File: xcube/core/store/fs/store.py [hunk 2]
```diff
@@ -581,6 +581,8 @@ def _generate_data_ids(self,
                            return_tuples: bool,
                            current_depth: int):
         root = self.root + ('/' + dir_path if dir_path else '')
+        if not self.fs.exists(root):
+            return
```
Comment (AliceBalfanz) [inline xcube/core/store/fs/store.py:585]: Why return? where is the directory created in case it does not exist?

**Review by AliceBalfanz — CHANGES_REQUESTED** [review]
Pls also set the version to 9.2.0 dev :) Let's first test and if it solves all our issues, then we can make a release

File: xcube/core/store/fs/store.py (hunk 2 above)
Comment (TonioF) [reply xcube/core/store/fs/store.py:585]: Return because no data ids exist

**Comment by TonioF** [comment]
Yes, definitely let's first do a dev version.

**Pull request merged**